Source/CarthageKit/Project.swift

```diff
@@ -458,6 +462,31 @@ public final class Project {
 				return copyProduct(dSYMURL, resolvedDestinationURL)
 			}
 	}
+	
+	/// Copies any *.bcsymbolmap files matching the given framework and contained
+	/// within the given directory URL to the directory that the framework
+	/// resides within.
+	///
+	/// If no bcsymbolmap files are found for the given framework, completes with
+	/// no values.
+	///
+	/// Sends the URLs of the bcsymbolmap files after copying.
+	public func copyBCSymbolMapToBuildFolderForFramework(frameworkURL: NSURL, fromDirectoryURL directoryURL: NSURL) -> SignalProducer<NSURL, CarthageError> {
+		return UUIDsForFramework(frameworkURL)
+			|> flatMap(.Merge) { uuids in SignalProducer(values: uuids) }
+			|> flatMap(.Merge) { uuid in
+				let fileName = uuid.UUIDString.stringByAppendingPathExtension("bcsymbolmap")!
+				let fileURL = directoryURL.URLByAppendingPathComponent(fileName)
+				if fileURL.checkResourceIsReachableAndReturnError(nil) {
```

Comment:
Could you factor out the logic above into separate functions as same with `dSYMsInDirectory()` and `dSYMForFramework()`?